src/middlewared/middlewared/plugins/etc.py

```diff
@@ -125,9 +125,15 @@ def __init__(self, *args, **kwargs):
                 with open(outfile, 'rb') as f:
                     existing_hash = hashlib.sha256(f.read()).hexdigest()
                 new_hash = hashlib.sha256(rendered.encode('utf-8')).hexdigest()
-                if existing_hash == new_hash:
-                    self.logger.debug(f'No new changes for {outfile}')
-                    continue
+                if existing_hash != new_hash:
+                    with open(outfile, 'w') as f:
+                        f.write(rendered)
+                        changes = True
+
+            if not os.path.exists(outfile):
+                with open(outfile, 'w') as f:
+                    f.write(rendered)
+                changes = True
```

Comment:
We are writing the file again below, regardless.

Reply:
Apparently I am missing commits here. I'm trying to investigate but I can't reach redmine ;-)

Reply:
I see what is going on. My fixes for this are in another ticket. I can probably just close this one out and fix that one since it has a similar issue.

Reply:
Okay, nuking this one. I've fixed the issue and that code is in PR https://github.com/freenas/freenas/pull/1643. I need to make PR's for some of these fixes back for master.